managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -4275,7 +4275,14 @@ public void testReadEntriesWithFilterOut() throws ManagedLedgerException, Interr
         cursor.asyncReadEntriesWithSkipOrWait(sendNumber, new ReadEntriesCallback() {
             @Override
             public void readEntriesComplete(List<Entry> entries, Object ctx) {
-                completableFuture.complete(entries.size());
+                try {
+                    entries.forEach(entry -> {
+                        assertNotEquals(entry.getEntryId() % 2, 0);
```

Comment:
It should be `assertEquals(entry.getEntryId() % 2, 0);`

And due to this one is executed by the SafeRun, so the test will not get failed event if the assert fails.

Reply:
> It should be `assertEquals(entry.getEntryId() % 2, 0);`

It should be `assertSame(entry.getEntryId() % 2, 0);`

Reply:
> And due to this one is executed by the SafeRun, so the test will not get failed event if the assert fails.

I already use `completableFuture.completeExceptionally(e)` to receive exception, it should be can work.